### PR TITLE
Replace use of the TLS presentation language with QUIC encoding

### PR DESCRIPTION
**Pull request by DavidSchinazi**
This PR is purely editorial and does not change the protocol in any way.

When we discussed #3020 in Cupertino, it emerged that there was consensus for this editorial change but not for all topics in that issue, so I split this out to its own PR.

### CODE REVIEW

**Comment by mikkelfj** [comment]
So the variable length integers have a 2 byte length prefix followed by a two bit length encoding?

File: draft-ietf-quic-tls.md [hunk 1]
```diff
@@ -1336,7 +1336,7 @@ protocol incompatible with the protocol version being used.
 ## QUIC Transport Parameters Extension {#quic_parameters}
 
 QUIC transport parameters are carried in a TLS extension. Different versions of
-QUIC might define a different format for this struct.
+QUIC might define a different format for them.
```
Comment (martinthomson) [inline draft-ietf-quic-tls.md]: "them" is a little vague.

```suggestion
QUIC might define a different method for negotiating transport configuration.
```

File: draft-ietf-quic-transport.md [hunk 2]
```diff
@@ -4347,42 +4347,47 @@ connection.
...
+parameters, in bytes. Each transport parameter is encoded as an (identifier,
+length, value) tuple, as shown in {{transport-parameter-encoding-fig}}:
+
+~~~
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|  Transport Parameter ID (16)  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|  Transport Param Length (16)  |
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md]: This should be a multi-line suggestion, but thanks to GitHub, it doesn't work that way...

```suggestion
|  Transport Parameter ID (16)  |  Transport Param Length (16)  |
```

File: draft-ietf-quic-transport.md [hunk 3]
```diff
@@ -4522,21 +4527,43 @@ preferred_address (0x000d):
...
-     opaque connectionId<0..20>;
-     opaque statelessResetToken[16];
-   } PreferredAddress;
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                       IPv4 Address (32)                       |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|         IPv4 Port (16)        |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md]: You could wrap this differently...

```
+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
|         IPv4 Port (16)        |                               |
+-------------------------------+                               +
|                                                               |
+                                                               +
|                      IPv6 Address (128)                       |
+                                                               +
|                                                               |
+                               +-------------------------------+
|                               |         IPv6 Port (16)        |
+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```

File: draft-ietf-quic-transport.md [hunk 4]
```diff
@@ -4522,21 +4527,43 @@ preferred_address (0x000d):
...
++                      IPv6 Address (128)                       +
+|                                                               |
++                                                               +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|         IPv6 Port (16)        |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+| CID Length (8)|
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                      Connection ID (*)                      ...
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md]:
```suggestion
| CID Length (8)|      Connection ID (*)                      ...
```

File: draft-ietf-quic-transport.md (hunk 3 above)
Comment (DavidSchinazi) [reply draft-ietf-quic-transport.md]: That looks sad :(

File: draft-ietf-quic-transport.md (hunk 4 above)
Comment (DavidSchinazi) [reply draft-ietf-quic-transport.md]: I'd rather follow the format we've been using in the definition of long headers

**Comment by rpaulo** [comment]
Are we doing https://github.com/quicwg/base-drafts/issues/2944 here or not?

**Comment by DavidSchinazi** [comment]
@rpaulo this is an editorial PR that does not change any wire encoding. We are not doing #2944 here.

**Comment by rpaulo** [comment]
This  not marked as editorial.  It was decided in Cupertino to either do #2944 here or to close #2944.

**Comment by DavidSchinazi** [comment]
@rpaulo that is not my recollection. Either way, this is a PR, we can discuss non-editorial changes on the appropriate issue.

File: draft-ietf-quic-transport.md (hunk 3 above)
Comment (kazuho) [reply draft-ietf-quic-transport.md]: Seriously, I believe the diagram should be left as-is.

IIUC, our convention is to wrap the lines in cases where the following field is bigger than 32-bits. Or do we want to wrap the Version field of long header packets?

**Pull request merged**